upup/pkg/fi/cloudup/template_functions.go

```diff
@@ -327,6 +327,12 @@ func (tf *TemplateFunctions) AddTo(dest template.FuncMap, secretStore fi.SecretS
 	dest["KarpenterInstanceTypes"] = func(ig kops.InstanceGroupSpec) ([]string, error) {
 		return karpenterInstanceTypes(tf.cloud.(awsup.AWSCloud), ig)
 	}
+	dest["KarpenterTTLSecondsAfterEmpty"] = func(ig kops.InstanceGroupSpec) int64 {
+		if ig.MaxInstanceLifetime != nil {
+			return int64(ig.MaxInstanceLifetime.Seconds())
+		}
+		return 300
+	}
```

Comment:
I don't think this mapping is good. `MaxInstanceLifetime` has nothing to do with this.
We should just set `ttlSecondsAfterEmpty: 300` or `180`.

Reply:
It was a mistake on my end. I confused this one with `ttlSecondsUntilExpired`